input-explainer.md

```diff
@@ -230,22 +235,18 @@ function onSelectStart(event) {
   // Ignore the event if we are already dragging
   if (activeDragInteraction)
     return;
-  
-  let inputPose = event.frame.getInputPose(event.inputSource, xrReferenceSpace);
-  // Ignore the event if this input source is not capable of tracking.
-  if (!inputPose || !inputPose.gripTransform)
-    return;
+
+  let targetRayPose = event.frame.getPose(event.inputSource.targetRaySpace, xrReferenceSpace);
 
   // Use the input source target ray to find a draggable object in the scene
-  let ray = inputPose.targetRay;
-  let hitResult = scene.hitTest(ray.origin, ray.direction);
+  let hitResult = scene.hitTest(new XRRay(targetRayPose.transform));
   if (hitResult && hitResult.draggable) {
-    // Use the grip position to drag the intersected object, rather than the target ray.
+    // Use the targetRayPose position to drag the intersected object.
```

Comment:
Is the change from dragging at the grip pose to dragging at the target ray origin intentional here?  Is the idea to support dragging with a broader set of input sources that may not have grip poses?

In practice, I wonder if 3DoF VR controllers that provide an emulated "shoot from the hip" target ray origin would supply an emulated grip pose as well, if only to allow for rendering of a controller model that matches the emulated target ray.

Reply:
Note that for a "shoot from the hip" 3DoF controller that only has a target ray and no grip pose, the target ray origin is not going to be very useful for dragging either, since it will basically be at the hip as well, and the furthest the user could drag an object by moving their hand would be the radius of the controller.

In that case, I'd suggest that conditioning on the availability of `gripSpace` here is more correct for dragging - if a controller doesn't have a grip pose, it's almost certainly not going to give you enough range of target ray origin 6DoF motion to actually perform a drag operation.  The app would need to implement some alternate dragging interaction for that case, perhaps based on the end of the target ray rather than its origin.